config/v1/types.go

```diff
@@ -310,3 +310,91 @@ type DelegatedAuthorization struct {
+	// the route must specify the HSTS Policy components described in the matching RequiredHSTSPolicy.
+	// If domainPatterns is empty, the specified HSTS Policy components are IGNORED.
+	//
+	// The use of wildcards is allowed like this: *.foo.com matches everything under foo.com.
+	// foo.com only matches foo.com, so to cover foo.com and everything under it, you must specify *both*.
+	// +optional
+	DomainPatterns []string `json:"domainPatterns,omitempty" protobuf:"bytes,2,rep,name=domainsPattern"`
+
+	// maxAge is the delta time range in seconds during which hosts are regarded as HSTS hosts.
+	// If set to 0, it negates the effect, and hosts are removed as HSTS hosts.
+	// If set to 0 and includeSubdomains is specified, all subdomains of the host are also removed as HSTS hosts.
+	// maxAge is a time-to-live value, and if this policy is not refreshed on a client, the HSTS
+	// policy will eventually expire on that client.
+	// +required
+	MaxAge MaxAgePolicy `json:"maxage" protobuf:"bytes,3,opt,name=maxage"`
+
+	// preloadPolicy directs the client to include hosts in its host preload list so that
+	// it never needs to do an initial load to get the HSTS header (note that this is not defined
+	// in RFC 6797 and is therefore client implementation-dependent).
+	// +optional
+	PreloadPolicy PreloadPolicy `json:"preloadPolicy,omitempty" protobuf:"bytes,4,opt,name=preloadPolicy"`
+
+	// includeSubDomainsPolicy means the HSTS Policy should apply to any subdomains of the host's
+	// domain name.  Thus, for the host bar.foo.com, if includeSubDomainsPolicy was set to RequireIncludeSubDomains:
+	// - the host app.bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host bar.foo.com would inherit the HSTS Policy of bar.foo.com
+	// - the host foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// - the host def.foo.com would NOT inherit the HSTS Policy of bar.foo.com
+	// +optional
+	IncludeSubDomainsPolicy IncludeSubDomainsPolicy `json:"includeSubDomainsPolicy,omitempty" protobuf:"bytes,5,opt,name=includeSubDomainsPolicy"`
+}
+
+// MaxAgePolicy contains a range for specifying a compliant HSTS max-age for the enclosing RequiredHSTSPolicy
+type MaxAgePolicy struct {
+	// The largest value of the RequiredHSTSPolicy max-age
+	// -1 means no opinion
+	// kubebuilder:validation:default=-1
+	LargestMaxAge int32 `json:"largestMaxAge,omitempty" protobuf:"varint,1,opt,name=largestMaxAge"`
+
+	// The smallest value of the RequiredHSTSPolicy max-age
+	// To set max-age=0 allows the deletion of an existing HSTS header from a host.  This is a necessary
+	// tool for administrators to quickly correct mistakes.  The minimum `max-age` default is -1 instead of 0.
+	// If it is set to 1 in MaxAgePolicy, then the HSTS policy header may not be inactivated unless it expires.
+	// -1 means no opinion.
+	// kubebuilder:validation:default=-1
+	SmallestMaxAge int32 `json:"smallestMaxAge,omitempty" protobuf:"varint,1,opt,name=largestMaxAge"`
```

Comment:
The doc on this field reads like a very compelling reason to have a `*int32` where nil and 0 mean different things.  How do you feel about allowing nil and instead of `-1`, "I want the default" is expressed as nil?

Reply:
I saw it before too.  I'm still thinking about this.

Reply:
I agree -- will use nil instead of -1 on this.  Does it sound right to make the default smallest max = 1 day and largest max = 1 year?

Reply:
I decided it doesn't make sense to default these values.  If they are missing, it indicates that the admin doesn't care.